appengine/genetics/js/blocks.js

```diff
@@ -0,0 +1,465 @@
+  });
+};
+
+/**
+ * Block for defining mouse decision on which other mouse to fight.
+ * @type {{init: !Function}}
+ */
+Blockly.Blocks['genetics_pickFight'] = {
+  init: goog.partial(Blockly.Blocks['genetics_mouseFunctionInit_'],
+                     'pickFight', '', 'Mouse')
+};
+
+/**
+ * Defines the JavaScript generation for pickFight.
+ * @type {!Function}
+ */
+Blockly.JavaScript['genetics_pickFight'] =
+    goog.partial(Blockly.JavaScript['genetics_mouseFunction_'],
+                 'pickFight', '');
+/**
+ * Block for defining mouse decision on which other mouse to mate with.
+ * @type {{init: !Function}}
+ */
+Blockly.Blocks['genetics_chooseMate'] = {
+  init: goog.partial(Blockly.Blocks['genetics_mouseFunctionInit_'],
+                     'chooseMate', '', 'Mouse')
+};
+
+/**
+ * Defines the JavaScript generation for chooseMate.
+ * @type {!Function}
+ */
+Blockly.JavaScript['genetics_chooseMate'] =
+    goog.partial(Blockly.JavaScript['genetics_mouseFunction_'],
+                 'chooseMate', '');
+/**
+ * Block for defining mouse decision on whether to mate with a specific mouse.
+ * @type {{init: !Function}}
+ */
+Blockly.Blocks['genetics_mateAnswer'] = {
+  init: goog.partial(Blockly.Blocks['genetics_mouseFunctionInit_'],
+                     'mateAnswer', 'suitor', 'Boolean'),
+  /**
+   * Return all variables referenced by this block.
+   * @return {!Array.<string>} List of variable names.
+   * @this Blockly.Block
+   */
+  getVars: function() {
+    return ['suitor'];
+  },
+  /**
+   * Add custom menu options to this block's context menu.
+   * @param {!Array} options List of menu options to add to.
+   * @this Blockly.Block
+   */
+  customContextMenu: function(options) {
+    // Add options to create getters for suitor parameter.
+    if (!this.isCollapsed()) {
+      var option = {enabled: true};
+      var name = 'suitor';
+      option.text = Blockly.Msg.VARIABLES_SET_CREATE_GET.replace('%1', name);
+      var xmlField = goog.dom.createDom('field', null, name);
+      xmlField.setAttribute('name', 'VAR');
+      var xmlBlock = goog.dom.createDom('block', null, xmlField);
+      xmlBlock.setAttribute('type', 'variables_get');
+      option.callback = Blockly.ContextMenu.callbackFactory(this, xmlBlock);
+      options.push(option);
+    }
+  }
+};
+
+/**
+ * Defines the JavaScript generation for mateAnswer.
+ * @type {!Function}
+ */
+Blockly.JavaScript['genetics_mateAnswer'] =
+    goog.partial(Blockly.JavaScript['genetics_mouseFunction_'],
+                 'mateAnswer', 'suitor');
+
+Blockly.Blocks['genetics_getSelf'] = {
+  /**
+   * Block for getting the mouse making the decision.
+   * @this Blockly.Block
+   */
+  init: function() {
+    this.jsonInit({
+      "message0": "%1(%2)",
+      "args0": ["getSelf", ""],
+      "output": "Mouse",
+      "colour": Genetics.Blocks.GENETICS_HUE,
+      "tooltip": BlocklyGames.getMsg('Genetics_meTooltip')
```

Comment:
Can the tooltip identifier strictly follow the name, in order to avoid unnecessary translation? I.e. Genetics_getSelfTooltip